app/gui/src/controller/searcher.rs

```diff
@@ -1235,16 +1244,63 @@ impl EditMetadataGuard {
             }),
         )
     }
+
+    fn get_saved_expression(&self) -> FallibleResult<Option<NodeEditStatus>> {
+        let module = &self.graph.graph().module;
+        let mut edit_status = None;
+        module.with_node_metadata(
+            self.node_id,
+            Box::new(|m| {
+                edit_status = m.edit_status.clone();
+            }),
+        )?;
+        Ok(edit_status)
+    }
+
+    fn revert_node_expression(&self) -> FallibleResult {
+        let edit_status = self.get_saved_expression()?;
+        match edit_status {
+            None => {
+                tracing::warn!("Tried to revert edited node, but found not edit metadata.");
```

Comment:
Please check the sentences with gramarly or something similar. This should be "... , but no edit metadata was found.". Also, what does it mean "revert edited node". You are not reverting the node, rather reverting its expression to the previous state, right? For example, node icons are not reverted? I believe that naming / explanations should be improved here and possibly in other places too.